DOC: Add py:type reference targets to pandas.api.typing.aliases (#63812)
pandas-dev/pandas: doc/source/reference/aliases.rst

Each of these aliases listed in the table below can be found by importing them from :py:mod:`pandas.api.typing.aliases`.

.. list-table::
   :header-rows: 1
   :widths: 30 70

   * - Alias
     - Meaning
   * - .. type:: AggFuncType

>      - Type of functions that can be passed to :meth:`DataFrame's <pandas.DataFrame.agg>`, :meth:`Series' <pandas.Series.agg>`, and :meth:`DataFrameGroupBy's <pandas.api.typing.DataFrameGroupBy.aggregate>` aggregate methods

Reviewer: ```suggestion
     - Type of functions that can be passed to :meth:`DataFrame's <pandas.DataFrame.agg>`, :meth:`Series' <pandas.Series.agg>`, and :meth:`DataFrameGroupBy's <pandas.api.typing.DataFrameGroupBy.aggregate>` ``aggregate()`` methods
```

   * - .. type:: AlignJoin
     - Argument type for ``join`` in :meth:`DataFrame's <pandas.DataFrame.align>` and :meth:`Series' <pandas.Series.align>` ``align`` method
   * - .. type:: AnyAll

>      - Argument type for ``how`` in :meth:`dropna`

Reviewer: create links to `Index.dropna()`, `Series.dropna()` and `DataFrame.dropna()`

   * - .. type:: AnyArrayLike
     - Used to represent :class:`~pandas.api.extensions.ExtensionArray`, ``numpy`` arrays, :class:`Index` and :class:`Series`
   * - .. type:: ArrayLike
     - Used to represent :class:`~pandas.api.extensions.ExtensionArray`, ``numpy`` arrays
   * - .. type:: AstypeArg

>      - Argument type in :meth:`astype`

Reviewer: create links to `Series.astype()`, `DataFrame.astype()`, and `Index.astype()`

>      - Argument type in :meth:`DataFrame's <pandas.DataFrame.astype>` and :meth:`Series' <pandas.Series.astype>` ``astype`` method

Reviewer: ```suggestion
     - Argument type in :meth:`DataFrame's <pandas.DataFrame.astype>` and :meth:`Series' <pandas.Series.astype>` ``astype()`` methods
```

   * - .. type:: Axes
     - :py:type:`AnyArrayLike` plus sequences (not strings) and ``range``
   * - .. type:: Axis
     - Argument type for ``axis`` in many methods
   * - .. type:: CSVEngine
     - Argument type for ``engine`` in :meth:`pandas.read_csv`
   * - .. type:: ColspaceArgType
     - Argument type for ``colspace`` in :meth:`pandas.DataFrame.to_html`
   * - .. type:: CompressionOptions
     - Argument type for ``compression`` in all I/O output methods except :meth:`pandas.DataFrame.to_parquet`
   * - .. type:: CorrelationMethod

>      - Argument type for ``correlation`` in :meth:`DataFrame's <pandas.DataFrame.corr>` and :meth:`Series' <pandas.Series.corr>` ``corr`` method

Reviewer: ```suggestion
     - Argument type for ``correlation`` in :meth:`DataFrame's <pandas.DataFrame.corr>` and :meth:`Series' <pandas.Series.corr>` ``corr()`` methods
```

   * - .. type:: DropKeep
     - Argument type for ``keep`` in :meth:`DataFrame's <pandas.DataFrame.drop_duplicates>` and :meth:`Series' <pandas.Series.drop_duplicates>` ``drop_duplicates`` method
   * - .. type:: Dtype
     - Types as objects that can be used to specify dtypes
   * - .. type:: DtypeArg
     - Argument type for ``dtype`` in various methods
   * - .. type:: DtypeBackend
     - Argument type for ``dtype_backend`` in various methods
   * - .. type:: DtypeObj
     - Numpy dtypes and Extension dtypes
   * - .. type:: ExcelWriterIfSheetExists
     - Argument type for ``if_sheet_exists`` in :class:`~pandas.ExcelWriter`
   * - .. type:: ExcelWriterMergeCells

>      - Argument type for ``merge_cells`` in :meth:`DataFrame's <pandas.DataFrame.to_excel>` and :meth:`Series' <pandas.Series.to_excel>` ``to_excel`` method

Reviewer: ```suggestion
     - Argument type for ``merge_cells`` in :meth:`DataFrame's <pandas.DataFrame.to_excel>` and :meth:`Series' <pandas.Series.to_excel>` ``to_excel()`` methods
```

   * - .. type:: FilePath
     - Type of paths for files for I/O methods
   * - .. type:: FillnaOptions
     - Argument type for ``method`` in various methods where NA values are filled
   * - .. type:: FloatFormatType

>      - Argument type for ``float_format`` in :meth:`DataFrame's <pandas.DataFrame.to_string>` and :meth:`Series' <pandas.Series.to_string>` ``to_string`` method

Reviewer: ```suggestion
     - Argument type for ``float_format`` in :meth:`DataFrame's <pandas.DataFrame.to_string>` and :meth:`Series' <pandas.Series.to_string>` ``to_string()`` methods
```

   * - .. type:: FormattersType
     - Argument type for ``formatters`` in :meth:`DataFrame's <pandas.DataFrame.to_string>` and :meth:`Series' <pandas.Series.to_string>` ``to_string`` method
   * - .. type:: FromDictOrient
     - Argument type for ``orient`` in :meth:`~pandas.DataFrame.from_dict`
   * - .. type:: HTMLFlavors
     - Argument type for ``flavor`` in :meth:`pandas.read_html`
   * - .. type:: IgnoreRaise
     - Argument type for ``errors`` in multiple methods
   * - .. type:: IndexLabel
     - Argument type for ``level`` in multiple methods
   * - .. type:: InterpolateOptions

>      - Argument type for ``interpolate`` in :meth:`DataFrame's <pandas.DataFrame.interpolate>` and :meth:`Series' <pandas.Series.interpolate>` ``interpolate`` method

Reviewer: ```suggestion
     - Argument type for ``interpolate`` in :meth:`DataFrame's <pandas.DataFrame.interpolate>` and :meth:`Series' <pandas.Series.interpolate>` ``interpolate()`` methods
```

   * - .. type:: JSONEngine
     - Argument type for ``engine`` in :meth:`read_json`
   * - .. type:: JSONSerializable
     - Argument type for the return type of a callable for argument ``default_handler`` in :meth:`to_json`
   * - .. type:: JoinHow
     - Argument type for ``how`` in :meth:`pandas.merge_ordered` and for ``join`` in :meth:`Series.align`
   * - .. type:: JoinValidate
     - Argument type for ``validate`` in :meth:`~pandas.DataFrame.join`
   * - .. type:: MergeHow
     - Argument type for ``how`` in :meth:`merge`
   * - .. type:: MergeValidate

>      - Argument type for ``validate`` in :meth:`merge`

Reviewer: create links to `DataFrame.merge()` and `pandas.merge()`

>      - Argument type for ``validate`` in :meth:`~pandas.DataFrame.join`

Reviewer: ```suggestion
     - Argument type for ``validate`` in :meth:`DataFrame.join() <pandas.DataFrame.join>`
```

   * - .. type:: NaPosition

>      - Argument type for ``na_position`` in :meth:`DataFrame's <pandas.DataFrame.sort_values>` and :meth:`Series' <pandas.Series.sort_values>` ``sort_values`` method

Reviewer: ```suggestion
     - Argument type for ``na_position`` in :meth:`DataFrame's <pandas.DataFrame.sort_values>` and :meth:`Series' <pandas.Series.sort_values>` ``sort_values()`` methods
```

   * - .. type:: NsmallestNlargestKeep

>      - Argument type for ``keep`` in :meth:`DataFrame's <pandas.DataFrame.nlargest>` and :meth:`Series' <pandas.Series.nlargest>` ``nlargest``, :meth:`DataFrame's <pandas.DataFrame.nsmallest>` and :meth:`Series' <pandas.Series.nsmallest>` ``nsmallest``, and :meth:`SeriesGroupBy's <pandas.core.groupby.SeriesGroupBy.nlargest>` ``nlargest`` methods

Reviewer: ```suggestion
     - Argument type for ``keep`` in :meth:`DataFrame's <pandas.DataFrame.nlargest>` and :meth:`Series' <pandas.Series.nlargest>` ``nlargest``, :meth:`DataFrame's <pandas.DataFrame.nsmallest>` and :meth:`Series' <pandas.Series.nsmallest>` ``nsmallest``, and :meth:`SeriesGroupBy's <pandas.api.typing.SeriesGroupBy.nlargest>` ``nlargest`` methods
```

   * - .. type:: OpenFileErrors
     - Argument type for ``errors`` in :meth:`to_hdf` and :meth:`to_csv`
   * - .. type:: Ordered

>      - Return type for :py:attr:`ordered` in :class:`CategoricalDtype` and :class:`Categorical`

Reviewer: fix links here

   * - .. type:: ParquetCompressionOptions
     - Argument type for ``compression`` in :meth:`~pandas.DataFrame.to_parquet`
   * - .. type:: QuantileInterpolation
     - Argument type for ``interpolation`` in :meth:`DataFrame's <pandas.DataFrame.quantile>` and :meth:`Series' <pandas.Series.quantile>` ``quantile`` method
   * - .. type:: ReadBuffer
     - Additional argument type corresponding to buffers for various file reading methods
   * - .. type:: ReadCsvBuffer
     - Additional argument type corresponding to buffers for :meth:`pandas.read_csv`
   * - .. type:: ReadPickleBuffer
     - Additional argument type corresponding to buffers for :meth:`pandas.read_pickle`
   * - .. type:: ReindexMethod

>      - Argument type for ``reindex`` in :meth:`reindex`

Reviewer: create links to `DataFrame.reindex()`, `Series.reindex()` and `Index.reindex()`

>      - Argument type for ``interpolation`` in :meth:`DataFrame's <pandas.DataFrame.quantile>` and :meth:`Series' <pandas.Series.quantile>` ``quantile`` method

Reviewer: ```suggestion
     - Argument type for ``interpolation`` in :meth:`DataFrame's <pandas.DataFrame.quantile>` and :meth:`Series' <pandas.Series.quantile>` ``quantile()`` methods
```

   * - .. type:: Scalar

>      - Types that can be stored in :class:`Series` with non-object dtype

Reviewer: fix link

>      - Argument type for ``reindex`` in :meth:`DataFrame's <pandas.DataFrame.reindex>` and :meth:`Series' <pandas.Series.reindex>` ``reindex`` method

Reviewer: ```suggestion
     - Argument type for ``reindex`` in :meth:`DataFrame's <pandas.DataFrame.reindex>` and :meth:`Series' <pandas.Series.reindex>` ``reindex()`` methods
```

   * - .. type:: SequenceNotStr
     - Used for arguments that require sequences, but not plain strings
   * - .. type:: SliceType
     - Argument types for ``start`` and ``end`` in :meth:`Index.slice_locs`
   * - .. type:: SortKind
     - Argument type for ``kind`` in :meth:`DataFrame's <pandas.DataFrame.sort_values>` and :meth:`Series' <pandas.Series.sort_values>` ``sort_values`` method
   * - .. type:: StorageOptions
     - Argument type for ``storage_options`` in various file output methods
   * - .. type:: Suffixes
     - Argument type for ``suffixes`` in :meth:`merge`, :meth:`compare` and :meth:`merge_ordered`
   * - .. type:: TakeIndexer

>      - Argument type for ``indexer`` and ``indices`` in :meth:`take`

Reviewer: fix link

   * - .. type:: TimeAmbiguous
     - Argument type for ``ambiguous`` in time operations
   * - .. type:: TimeGrouperOrigin

>      - Argument type for ``origin`` in :meth:`DataFrame's <pandas.DataFrame.resample>`, :meth:`Series' <pandas.Series.resample>` and :class:`~pandas.Grouper`

Reviewer: ```suggestion
     - Argument type for ``origin`` in :meth:`DataFrame's <pandas.DataFrame.resample>`, :meth:`Series' <pandas.Series.resample>` ``resample` methods and for :class:`~pandas.Grouper`
```

   * - .. type:: TimeNonexistent
     - Argument type for ``nonexistent`` in time operations
   * - .. type:: TimeUnit
     - Time unit argument and return type for :attr:`~pandas.Timedelta.unit`, arguments ``unit`` and ``date_unit``
   * - .. type:: TimedeltaConvertibleTypes
     - Argument type for ``offset`` in various methods, such as :meth:`DataFrame's <pandas.DataFrame.resample>` and :meth:`Series' <pandas.Series.resample>` ``resample()``, ``halflife`` in :meth:`DataFrame's <pandas.DataFrame.ewm>`, :meth:`DataFrameGroupBy's <pandas.api.typing.DataFrameGroupBy.ewm>`, and :meth:`Series' <pandas.Series.ewm>` ``ewm()``, and ``start`` and ``end`` in :meth:`pandas.timedelta_range`
   * - .. type:: TimestampConvertibleTypes
     - Argument type for ``origin`` in :meth:`DataFrame's <pandas.DataFrame.resample>` and :meth:`Series' <pandas.Series.resample>` ``resample()``, and in :meth:`pandas.to_datetime`
   * - .. type:: ToStataByteorder
     - Argument type for ``byteorder`` in :meth:`~pandas.DataFrame.to_stata`
   * - .. type:: ToTimestampHow

>      - Argument type for ``how`` in :meth:`DataFrame's <pandas.DataFrame.to_timestamp>` and :meth:`Series' <pandas.Series.to_timestamp>` ``to_timestamp`` method, and ``convention`` in :meth:`DataFrame's <pandas.DataFrame.resample>` and :meth:`Series' <pandas.Series.resample>` ``resample()``

Reviewer: ```suggestion
     - Argument type for ``how`` in :meth:`DataFrame's <pandas.DataFrame.to_timestamp>` and :meth:`Series' <pandas.Series.to_timestamp>` ``to_timestamp()`` methods, and ``convention`` in :meth:`DataFrame's <pandas.DataFrame.resample>` and :meth:`Series' <pandas.Series.resample>` ``resample()`` methods
```

   * - .. type:: UpdateJoin
     - Argument type for ``join`` in :meth:`~pandas.DataFrame.update`
   * - .. type:: UsecolsArgType
     - Argument type for ``usecols`` in :meth:`pandas.read_clipboard`, :meth:`pandas.read_csv` and :meth:`pandas.read_excel`
   * - .. type:: WindowingRankType

>      - Argument type for ``method`` in :meth:`DataFrame's <pandas.DataFrame.rank>` and :meth:`Series' <pandas.Series.rank>` ``rank()`` in rolling and expanding window operations

Reviewer: ```suggestion
     - Argument type for ``method`` in :meth:`Rolling's <pandas.api.typing.Rolling.rank>` and :meth:`Expanding <pandas.api.typing.Expanding.rank>` ``rank()`` methods, applicable in rolling and expanding window operations
```

   * - .. type:: WriteBuffer
     - Additional argument type corresponding to buffers for various file output methods
   * - .. type:: WriteExcelBuffer

>      - Additional argument type corresponding to buffers for :meth:`DataFrame's <pandas.DataFrame.to_excel>` and :meth:`Series' <pandas.Series.to_excel>` ``to_excel`` method

Reviewer: ```suggestion
     - Additional argument type corresponding to buffers for :meth:`DataFrame's <pandas.DataFrame.to_excel>`, :meth:`Series' <pandas.Series.to_excel>`  and :meth:`Styler's <pandas.io.formats.style.Styler.to_excel>` ``to_excel()`` methods
```

   * - .. type:: XMLParsers

>      - Argument type for ``parser`` in :meth:`~pandas.DataFrame.to_xml` and :meth:`pandas.read_xml`

Reviewer: ```suggestion
     - Argument type for ``parser`` in :meth:`DataFrame.to_xml() <pandas.DataFrame.to_xml>` and :meth:`pandas.read_xml`
```
